src/NUnitFramework/tests/Internal/EventListenerTextWriterTests.cs

```diff
@@ -25,24 +25,34 @@
 using System.Collections.Generic;
 using System.IO;
 using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal.Execution;
 
 namespace NUnit.Framework.Internal
 {
     [TestFixture]
     public class EventListenerTextWriterTests
     {
+        private static readonly string STREAM_NAME = "EventListenerTextWriterTestsStream";
         private static readonly string NL = Environment.NewLine;
 
-        TestListenerHelper TestResult;
-        TextWriter TestWriter;
+        TestListenerIntercepter ListenerResult;
+        TextWriter ListenerWriter;
 
         [SetUp]
         public void SetUp()
         {
-            TestResult = new TestListenerHelper();
-            TestExecutionContext.CurrentContext.Listener = TestResult;
+            // Wrap the current listener, listening to events, and forwarding the original event
+            ListenerResult = new TestListenerIntercepter(TestExecutionContext.CurrentContext.Listener);
+            TestExecutionContext.CurrentContext.Listener = ListenerResult;
 
-            TestWriter = TextWriter.Synchronized(TestContext.Error);
+            ListenerWriter = TextWriter.Synchronized(new EventListenerTextWriter(STREAM_NAME, TextWriter.Null));
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            // Restore the original listener
+            TestExecutionContext.CurrentContext.Listener = ListenerResult.DefaultListener;
```

Comment:
Of course, the context is about to be destroyed anyway, but I think this is good practice.